### DISCUSSION TITLE
How to load URl dynamically on a <ui:include without reloading

### DISCUSSION THREAD

**Discussion by Moselan** [primefaces/community#961] [PrimeFaces] [answered]
I have this template

```
         <div id="ppal">
		<div id="cabecera">
			<ui:insert name="cabecera"><ui:include src="header.xhtml"/></ui:insert>
		</div>
		
  		<div id="menu">
			<ui:insert name="menu"><ui:include src="menu.xhtml"/></ui:insert>
		</div>
  		<div id="contenedor">			
  	  		 <ui:insert name="contenedor"> <ui:include src="#{Bean.pagina}" /></ui:insert>
		</div>	
		
		<div id="pie">
  			<ui:insert name="pie"><ui:include src="footer.xhtml"/></ui:insert>
		</div>
	</div>
```

This menu:

```
 <p:submenu> 
        	<p:menuitem actionListener="#{Bean.changePage('page1.xhtml')}" ajax="false">
        	<p:menuitem actionListener="#{Bean.changePage('page2.xhtml')}" ajax="false">
        	<p:menuitem actionListener="#{Bean.changePage('page2.xhtml')}" ajax="false">			
        </p:submenu>
```
        
   And this is the method to change Page on Bean:
   
 ```
  public void changePage( String page) {		
		this.page=pagina;	
	}
	
```

It works correctly, the problem I have is it reloads the entire web, not only load the div "contenedor"
Any way to achieve it withou reloading all?

**Comment by melloware** [comment]
Yes there is you can use AJAX and update those DIVs if you change them to `p:outputPanel` etc or use `p:fragment`.

See: https://www.primefaces.org/showcase/ui/ajax/fragment.xhtml?jfwid=9606b

See: https://www.primefaces.org/showcase/ui/ajax/observer.xhtml?jfwid=9606b

Plenty of great examples on how to `update` only a piece of the DOM and not reload the whole page.

**Comment by Rapster** [comment, marked as answer]
There might be a few basics that you should read up here. You specifically set ajax to false, so you shouldn't suprised it reloads the entire page:

https://stackoverflow.com/a/25347186

A simple search on google would have give you this:

https://stackoverflow.com/a/7113961

As you seem quite new to JSF: you should read up all of this https://jsf.zeef.com/bauke.scholtz 😉 

PrimeFaces is out of the equation here, basic component _(e.g `<h:panelGroup />`, `<ui:fragment />`, jsf using passthrough element)_ are the recommended approach here

**Comment by Moselan** [comment]
Thanks. I will take a look deeper
